app/elastic/AwsSigner.scala

```diff
@@ -0,0 +1,84 @@
+package elastic
+
+import java.net.URI
+
+import com.amazonaws.DefaultRequest
+import com.amazonaws.auth.AWS4Signer
+import com.amazonaws.auth.AWSCredentials
+import com.amazonaws.auth.BasicAWSCredentials
+import com.amazonaws.http.HttpMethodName
+import scala.collection.JavaConverters._
+import java.io.StringReader
+import java.io.ByteArrayInputStream
+import java.io.InputStream
+
+object AwsSigner {
+
+  def sing(method: String, url: String, headers: Seq[(String, String)], body: Option[String], secret: String, key: String) : Seq[(String, String)]  = {
```

Comment:
I think this method should be: `sign` and not `sing`, right?

Reply:
You are right. Should I fix it and do another PR?

Reply:
In fact, I don't know, because this PR is open for so long. Did you build a docker image from your branch? if so, could share it with us, meanwhile they don't merge your PR.